subsys/fs/nffs_fs.c

```diff
@@ -218,44 +222,44 @@ static int inode_to_dirent(struct nffs_inode_entry *inode,
 	return rc;
 }
 
-int fs_open(fs_file_t *zfp, const char *file_name)
+static int nffs_open(fs_file_t *zfp, const char *file_name)
 {
 	int rc;
 
 	k_mutex_lock(&nffs_lock, K_FOREVER);
```

Comment:
If you're already locking at the VFS level, do you need to lock here? or do you expect those APIs to be directly callable even with VFS?

Reply:
@carlescufi , In VFS layer we are only protecting the operations on mount point liek adding new fs or mounting it.

The normal file and directory operations are not protected at VFS layer, these need to be taken care by individual file systems depending their implementation.
The scope of this commit is adapt the existing file system interfaces with VFS.

I expect more changes to nffs_fs/fat_fs in upcoming "multiple instances of the same file system" commits.